interfaces/builtin/docker_support.go

```diff
@@ -532,6 +536,10 @@ ptrace (read, trace) peer=unconfined,
 # This grants raw access to device files and thus device ownership
 /dev/** mrwkl,
 @{PROC}/** mrwkl,
+
+# When kubernetes drives docker, it creates files in the container at arbitrary
```

Comment:
I think this is because default docker containers use `pivot_root`, and as such AppArmor doesn't see the "real" path.

Note that what the docker snap does is different than "default" docker, because we specifically only let it do a `chroot` and not a `pivot_root`: https://git.launchpad.net/~docker/+git/snap/tree/patches/snappy-real-chroot.patch
@jdstrand would applying that patch to whatever part of microk8s that uses docker remove the need for this arbitrary file locations?

Reply:
@anonymouse64 - that is a great observation and I've advised the k8s team to adjust their microk8s packaging to use your docker (I used a modified microk8s snap to test this interface), but they haven't yet (it's more than just this that is a little wonky). I don't know their plans for microk8s, but they did say they wanted to have their k8s worker nodes *not* ship docker, but instead use your snap. Until these questions are answered, I think we need to keep the rule here. However, I'm all for removing the rule down the line (since docker-support is a super-privileged interface, no one can deliver it via the store without us knowing).

Reply:
I see, it sounds then like the ideal thing for the microk8s snap is to set it up to use docker over the `docker-executables` content interface (and then also plug `docker-daemon` `docker` interface) that the docker snap slot's so that the k8s nodes end up talking to the `dockerd` confined in the docker snap. But yes, as pointed out docker-support is already super privileged and it can always be refined back later as this is implemented if there's a urgent need for microk8s (it seems like there is).